contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -82,182 +137,268 @@ contract MixinStakingPoolRewards is
-        // add dependencies on cumulative rewards for this epoch and the previous epoch, if necessary.
+        // Add dependencies on cumulative rewards for this epoch and the next
+        // epoch, if necessary.
         _setCumulativeRewardDependenciesForDelegator(
             poolId,
             finalDelegatedStakeToPoolByOwner,
             true
         );
 
-        // remove dependencies on previous cumulative rewards, if they are no longer needed.
+        // Remove dependencies on previous cumulative rewards, if they are no
+        // longer needed.
         _setCumulativeRewardDependenciesForDelegator(
             poolId,
             initialDelegatedStakeToPoolByOwner,
             false
         );
     }
 
-    /// @dev Handles a pool's reward. This will deposit the operator's reward into the Eth Vault and
-    /// the members' reward into the Staking Pool Vault. It also records the cumulative reward, which
-    /// is used to compute each delegator's portion of the members' reward.
+    /// @dev Handles a pool's reward at the current epoch.
+    ///      This will compute the reward split and record the cumulative
+    ///      reward, which is used to compute each delegator's portion of the
+    ///      members' reward. It will NOT make any transfers to the eth or
+    ///      reward vaults. That should be done with a separate call to
+    ///      `_depositStakingPoolRewards()``.
     /// @param poolId Unique Id of pool.
     /// @param reward received by the pool.
-    /// @param amountOfDelegatedStake the amount of delegated stake that will split the  reward.
-    /// @param epoch at which this was earned.
-    function _handleStakingPoolReward(
+    /// @param membersStake the amount of non-operator delegated stake that
+    ///        will split the  reward.
+    /// @return operatorReward
+    function _recordStakingPoolRewards(
         bytes32 poolId,
         uint256 reward,
-        uint256 amountOfDelegatedStake,
-        uint256 epoch
+        uint256 membersStake
     )
         internal
+        returns (uint256 operatorReward, uint256 membersReward)
     {
-        IStructs.Pool memory pool = poolById[poolId];
+        IStructs.Pool memory pool = _poolById[poolId];
 
-        // compute the operator's portion of the reward and transfer it to the ETH vault (we round in favor of the operator).
-        uint256 operatorPortion = amountOfDelegatedStake == 0
-            ? reward
-            : LibMath.getPartialAmountCeil(
-                uint256(pool.operatorShare),
-                PPM_DENOMINATOR,
-                reward
-            );
+        // Split the reward between operator and members
+        (operatorReward, membersReward) =
+            _splitStakingPoolRewards(pool.operatorShare, reward, membersStake);
 
-        ethVault.depositFor.value(operatorPortion)(pool.operator);
+        // Record the operator's reward in the eth vault.
+        ethVault.recordDepositFor(pool.operator, operatorReward);
```

Comment:
I kind of hate the `recordDepositFor` pattern. It creates the potential for vault balances to get out of sync and also adds some attack surface to the contracts. After Istanbul it also doesn't save _that_ much gas. Just making the deposits every time would add an additional ~1600 gas after [EIP-2200](https://github.com/ethereum/EIPs/pull/2200) right?

Even if it is a bit more expensive, I think the system clearly has higher security if we remove this pattern.